File: knowledgebase-generator/src/main/java/com/hartwig/hmftools/knowledgebasegenerator/transvar/TransvarOutput.java
package com.hartwig.hmftools.knowledgebasegenerator.transvar;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(allParameters = true,
             passAnnotations = { NotNull.class, Nullable.class })
public abstract class TransvarOutput {

    @NotNull
    public abstract String input();

    @NotNull
    public abstract String transcript();

    @NotNull
    public abstract String gene();

    @NotNull
    public abstract String strand();

    @NotNull
    public abstract String coordinates_gDNA_cDNA_protein();

    @NotNull
    public abstract String region();

    @NotNull
    public abstract String info();

}

File: knowledgebase-importer/src/main/kotlin/com/hartwig/hmftools/knowledgebaseimporter/transvar/TransvarOutput.kt
package com.hartwig.hmftools.knowledgebaseimporter.transvar

data class TransvarOutput(val gene: String, val coordinates: String, val region: String, val info: String) {
    companion object Factory {
        operator fun invoke(transvarOutput: String): TransvarOutput {
            val sections = transvarOutput.split("\t")
            // KODU: section[0] = input.
            // KODU: section[1] = transcript
            // KODU: section[3] = strand
            return TransvarOutput(sections[2], sections[4], sections[5], sections[6])
        }
    }
}
